lib/TPP/Conversion/ConvertVectorToXsmm/ConvertVectorToXsmm.cpp

```diff
@@ -0,0 +1,235 @@
+//===- ConvertVectorToXsmm.cpp ----------------------------------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "TPP/Conversion/ConvertVectorToXsmm/ConvertVectorToXsmm.h"
+#include "TPP/Dialect/Xsmm/XsmmUtils.h"
+#include "TPP/Transforms/Utils/BuilderUtils.h"
+#include "TPP/Transforms/Utils/VNNIUtils.h"
+#include "TPP/Transforms/Utils/ValueUtils.h"
+#include "mlir/Dialect/Func/IR/FuncOps.h"
+#include "mlir/Dialect/LLVMIR/LLVMDialect.h"
+#include "mlir/Dialect/MemRef/IR/MemRef.h"
+#include "mlir/Dialect/PDL/IR/PDL.h"
+#include "mlir/Dialect/PDLInterp/IR/PDLInterp.h"
+#include "mlir/Dialect/Transform/PDLExtension/PDLExtensionOps.h"
+#include "mlir/Dialect/Vector/IR/VectorOps.h"
+#include "mlir/IR/BuiltinDialect.h"
+#include "mlir/IR/Operation.h"
+#include "mlir/IR/Value.h"
+#include "mlir/Parser/Parser.h"
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
+#include "llvm/Support/Debug.h"
+using namespace mlir;
+using namespace mlir::vector;
+using namespace mlir::linalg;
+using namespace mlir::func;
+using namespace mlir::xsmm;
+
+#include "ConvertVectorToXsmmPDLLPatterns.h.inc"
+
+#define DEBUG_TYPE "convert-vector-to-xsmm"
+
+pair<Operation *, Operation *>
+getTransposeOperation(PatternRewriter &rewriter, xsmm::UnaryInfo &unaryInfo,
+                      Type outputType, xsmm::UnaryKind unaryKind,
+                      Operation *input, Operation *output,
+                      Operation *transposeOp) {
+  std::string dispatchName = "xsmm_unary_dispatch";
+  std::string invokeName = "xsmm_unary_invoke";
+  Location loc = transposeOp->getLoc();
+
+  ModuleOp module = transposeOp->getParentOfType<ModuleOp>();
+  auto dtype =
+      xsmm::utils::getDataType(rewriter, transposeOp->getOperand(0).getType());
+
+  auto functionOp = transposeOp->getParentOfType<func::FuncOp>();
+  SmallVector<Value, 10> dispatchOperands;
+  OpBuilder::InsertionGuard guard(rewriter);
+  rewriter.setInsertionPoint(&*functionOp.getBody().op_begin());
```

Comment:
This change of insertion point doesn't need to be this early. Move it down with the use (`buildXsmmCall`) so that it's clear why you're changing it.

Reply:
We'll then have to change the insertion point conditionally, as we're using buildXsmmCall for both invokes and dispatches, which I feel is confusing. What do you think?